core/dive.c

```diff
@@ -1208,13 +1208,26 @@ static void fixup_no_o2sensors(struct divecomputer *dc)
 	}
 }
 
+/* Remove sensor and gas pressure info */
+static void clear_sensor(struct sample *samples, int sample_id, int sensor_id)
+{
+	struct sample *s = &samples[sample_id];
+	if (sample_id == 0) {
+		s->sensor[sensor_id] = 0;
+		s->pressure[sensor_id].mbar = 0;
+	} else {
+		s->sensor[sensor_id] = s[-1].sensor[sensor_id];
+		s->pressure[sensor_id].mbar = s[-1].pressure[sensor_id].mbar;
+	}
+}
+
 static void fixup_dc_sample_sensors(struct divecomputer *dc, int nr_cylinders)
 {
 	for (int i = 0; i < dc->samples; i++) {
 		struct sample *s = dc->sample + i;
 		for (int j = 0; j < MAX_SENSORS; j++) {
 			if (s->sensor[j] < 0 || s->sensor[j] >= nr_cylinders)
-				s->sensor[j] = NO_SENSOR;
+				clear_sensor(dc->sample, i, j);
```

Comment:
See, here the old code made sense. It fixes up the one thing it says it should fix up. It does so badly, but still - sensibly.

The new code says it clears sensor data, but that's not really what it does.

Reply:
> See, here the old code made sense. It fixes up the one thing it says it should fix up. It does so badly, but still - sensibly.

The first problem is - with this code, we produce broken XML files.

The second problem is - I don't understand this part of the code _at all_.

Personally, my preferred solution would be to remove writing "legacy format" sensor data. But I don't understand why it is written in the first place, so I don't dare to remove it.

Help?